Frends.AzureBlobStorage.ListBlobsInContainer/Frends.AzureBlobStorage.ListBlobsInContainer/ListBlobsInContainer.cs

```diff
@@ -0,0 +1,168 @@
+﻿using System;
+using System.ComponentModel;
+using Azure.Storage.Blobs;
+using Azure;
+using Frends.AzureBlobStorage.ListBlobsInContainer.Definitions;
+using System.Threading.Tasks;
+using Azure.Storage.Blobs.Models;
+using System.Collections.Generic;
+using System.Threading;
+
+namespace Frends.AzureBlobStorage.ListBlobsInContainer
+{
+    public class AzureBlobStorage
+    {
+        /// <summary>
+        /// List blobs and subdirectories in Azure Storage container with flat or hierarchical listing structure.
+        /// [Documentation](https://tasks.frends.com/tasks/frends-tasks/Frends.AzureBlobStorage.ListBlobsInContainer)
+        /// </summary>
+        /// <param name="source">Source connection parameters.</param>
+        /// <param name="optional">Optional parameters</param>
+        /// <returns>object { string BlobType, string Uri, string Name, string ETag }</returns>
+
+        public static async Task<Result> ListBlobsInContainer([PropertyTab] Source source, [PropertyTab] Optional optional, CancellationToken cancellationToken)
+        {
+            var authSas = source.AuthenticationMethod.Equals(AuthenticationMethod.SASToken) ? true : false;
+            var blobContainerClient = CreateBlobContainerClient(source);
+            
+            if (optional.ListingStructure.Equals(ListingStructure.Flat))
+            {
+                var enumerable = blobContainerClient.GetBlobsAsync(BlobTraits.None, BlobStates.None, string.IsNullOrWhiteSpace(optional.Prefix) ? null : optional.Prefix).AsPages();
```

Comment:
Since this is an async method, you should pass cancellation token as a parameter.